src/site/topics/memory-callstack/memory-callstack.rst

```diff
@@ -3,88 +3,132 @@ Memory & The Call Stack
 ***********************
 
 
-* When running programs on our computer, memory is used and accessed in different ways
-* Typically, we physically store instructions *and* data in the same address space
-    * Note that there are other architectures/strategies one could use for a computational system
+* Programs use and access memory in different ways
+* Typically, instructions *and* data are physically stored in the same memory address space
+
+    * This is called a *Von Neumann* based architecture
+    * A noteworthy alternative is the *Harvard* architecture, which separates the instruction and data memory
+
+
+* Having a common memory address space for instructions and data could be a recipe for disaster
+* However, modern computer systems have strategies for mitigating potential issues
+* In many programming languages, Java included, memory is logically seperated into areas for
 
-* This could be a recipe for disaster, but our modern systems have strategies for mitigating this
-* For many programming languages, Java included, we logically separate out memory into areas for
     * Instructions
+
         * e.g. Classes & Interfaces
-    * Dynamically allocated stuff
-        * e.g. Objects
-    * Stuff related to what is currently happening
+
+
+    * Dynamically allocated data (memory being allocated at runtime)
+
+        * e.g. Instances of objects
+
+
+    * Program elements related to what is currently happening
+
         * e.g. Variables & parameters
 
+
+
 .. warning::
 
-    The content of this topic is kept at a high-level for introductory purposes. There may be many asterisks added to many of the
-    statements contained within.
+    This topic is intentionally kept at a high-level for the purpose of introducing these ideas. There are several
+    nuances and details related to statements contained within this topic that are skipped.
+
 
 
 Memory Allocation
 =================
 
-Memory is broken down into two broad sections:
+* Memory is broken down into two broad sections:
+
+
+#. The Stack
 
-1. The Stack
     * Stores information about the current method running
+
         * Local variables
         * Formal parameters
         * Return address
 
-2. The Heap
+
+#. The Heap
+
     * Which can be broken down further into storage for
+
 		* Static information
-		    * e.g. your code
+
+		    * e.g. Program code
```

Comment:
It's not exactly static. Classes that are never used are never loaded into memory. It's more like append only, until the program exits.

Sort of. Because JIT also exists.